extensions/wikia/ArticleComments/classes/ArticleComment.class.php

```diff
@@ -197,84 +192,124 @@ static public function newFromId( $id ) {
+			WikiaLogger::instance()->warning( 'Unable to load revision IDs', [
+				'issue' => 'SOC-1540',
+				'firstRevId' => $this->mFirstRevId,
+				'lastRevId' => $this->mLastRevId,
+				'title' => print_r( $this->mTitle, true ),
+			] );
+			return false;
+		}
 
-			if ( empty( $this->mLastRevId ) ) {
-				// assume article is bogus, treat as if it doesn't exist
-				wfProfileOut( __METHOD__ );
-				return false;
-			}
+		// Get revision objects
+		if ( !$this->loadFirstRevision() || !$this->loadLastRevision() ) {
+			WikiaLogger::instance()->warning( 'Unable to load revision objects', [
+				'issue' => 'SOC-1540',
+				'firstRevId' => $this->mFirstRevId,
+				'lastRevId' => $this->mLastRevId,
+				'title' => print_r( $this->mTitle, true ),
+			] );
+			return false;
+		}
+
+		// get user that created this comment
+		$this->mUser = User::newFromId( $this->mFirstRevision->getUser() );
+		$this->mUser->setName( $this->mFirstRevision->getUserText() );
 
+		$rawText = $this->mLastRevision->getText();
+		$this->parseText( $rawText );
+
+		$this->isLoaded = true;
+		return true;
+	}
+
+	/**
+	 * Load the first revision ID
+	 *
+	 * @param bool $useMaster
+	 *
+	 * @return bool Returns true if successful, false otherwise.
+	 */
+	private function loadFirstRevId( $useMaster = false ) {
+		if ( empty( $this->mTitle ) ) {
+			return false;
+		}
+
+		// Exit early if we've already loaded this
+		if ( !empty( $this->mFirstRevId ) ) {
+			return true;
+		}
+
+		if ( !$useMaster ) {
 			$this->mFirstRevId = $this->getFirstRevID( DB_SLAVE );
-			// if first rev does not exist on slave then fall back to master anyway
-			if ( !$this->mFirstRevId ) {
-				$this->mFirstRevId = $this->getFirstRevID( DB_MASTER );
-			}
+		}
 
-			if ( empty( $this->mFirstRevId ) ) {
-				// assume article is bogus, treat as if it doesn't exist
-				wfProfileOut( __METHOD__ );
-				return false;
-			}
+		// Fall back to master if not on slave or if we wanted master in the first place
+		if ( empty( $this->mFirstRevId ) ) {
+			$this->mFirstRevId = $this->getFirstRevID( DB_MASTER );
+		}
 
-			// get revision objects
-			if ( $this->mFirstRevId ) {
-				$this->mFirstRevision = Revision::newFromId( $this->mFirstRevId );
-				if ( !empty( $this->mFirstRevision ) && is_object( $this->mFirstRevision ) && ( $this->mFirstRevision instanceof Revision ) ) { // fix for FB:15198
-					if ( $this->mLastRevId == $this->mFirstRevId ) {
-						// save one db query by just setting them to the same revision object
-						$this->mLastRevision = $this->mFirstRevision;
-					} else {
-						$this->mLastRevision = Revision::newFromId( $this->mLastRevId );
-						if ( empty( $this->mLastRevision ) || !is_object( $this->mLastRevision ) || !( $this->mLastRevision instanceof Revision ) ) {
-							$result = false;
-						}
-					}
-				} else {
-					$result = false;
-				}
-			} else {
-				$result = false;
-			}
+		return !empty( $this->mFirstRevId );
```

Comment:
I'm not sure I understand the suggestion?  The idea is that this function will load member {{mFirstRevId}} with the correct value, and let the caller know if it succeeded.  The test for whether it succeeded is whether {{mFirstRevId}} is {{empty}} or not.  Are you suggesting something like this?
```php
$operationSuccessful = !empty( $this->mFirstRevId );
return $operationSuccessful;
```